### PR TITLE
Fix all Javadoc warnings

### PR DESCRIPTION
**Pull request by melissalinkert**
See https://github.com/ome/ome-common-java/issues/2

Without this PR, ```mvn javadoc:javadoc``` does not fail, but reports many warnings.  With this PR, the same command should continue to succeed and not produce any warnings.

There should not be any code changes here, only Javadoc changes.  I have not enabled ```mvn javadoc:javadoc``` in Travis, but could do so here if desired.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
Wow, this is a great improvement!

Regarding the `@author` fixes, would it be worth stripping them out of all the sources entirely?  They add almost zero value for the API reference.

Regarding sanity checking, could we make warnings and errors fatal in this PR, so that we avoid any future regressions if people forget to update the javadocs accompanying a change?

File: src/main/java/loci/common/Location.java [hunk 1]
```diff
@@ -629,9 +732,8 @@ public String getPath() {
   }
 
   /**
-   * Tests whether or not this path name is absolute.
```
[the remark this answers is not in the document]
Comment (melissalinkert) [reply src/main/java/loci/common/Location.java]: Should be fixed with f3ee3c3

**Comment by melissalinkert** [comment]
Removing ```<failOnError>false</failOnError>``` from the ```maven-javadoc-plugin``` configuration definitely isn't enough to fail the build on warnings.  I'll need to investigate more to see if there is some other way to do that.

I don't have a problem with removing ```@author``` tags in principle, but it's slightly outside the scope of what I was trying to do with this PR.  I can remove them here, but I'd feel better about doing so if we have a consensus that removing ```@author``` is fine in general (for all repos, and for all authors including  externals and former OME team).

**Comment by joshmoore** [comment]
> I can remove them here, but I'd feel better about doing so if we have a consensus that removing @author is fine in general (for all repos, and for all authors including externals and former OME team).

Independent of whether here or elsewhere, I'm :+1: for removing these method-author lines and have suggested (and started implementing) their removal from OMERO repos. Good point, though, that we should review what that means in terms of external contributors, etc.

**Comment by melissalinkert** [comment]
Further investigation of ```maven-javadoc-plugin``` indicates that there is no way to fail on warnings with 3.0.0, but that what we need will be included in 3.0.1:

https://github.com/apache/maven-javadoc-plugin/commit/f67d3a4940b76f189d072bc2e5d0a54335052006

I'd propose to defer POM and Travis changes until 3.0.1 is released, which probably means in a follow-up to this PR.

**Comment by mtbc** [comment]
3.0.1's out: omero-ms-mapr-jdbc uses it.

**Comment by mtbc** [comment]
Enabling in Travis sounds like a good idea.

**Pull request merged**